_pytest/config.py

```diff
@@ -942,6 +941,7 @@ def _consider_importhook(self, args, entrypoint_name):
         and find all the installed plugins to mark them for re-writing
         by the importhook.
         """
+        import pkg_resources
         ns, unknown_args = self._parser.parse_known_and_unknown_args(args)
         mode = ns.assertmode
         if mode == 'rewrite':
```

Comment:
I think it would be OK to move this to the `else` clause below, this would at least allow people to use `plain` if for some reason they don't have `pkg_resources`. OTOH if we really want to support running py.test without `pkg_resources`, we probably would need to write a test for it to ensure we don't break it again by accident.

@raquel-ucl is moving the import enough for you in Jython? It seems `_consider_importhook` function will be called regardless of the value of the `--assert=` option, so I think it will still break for you.

Btw, are you using `--assert=plain`?